sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkThriftServerProtocolVersionsSuite.scala

```diff
@@ -224,7 +224,9 @@ class SparkThriftServerProtocolVersionsSuite extends HiveThriftJdbcTest {
       }
       testExecuteStatementWithProtocolVersion(version, "SELECT cast(49960 as binary)") { rs =>
         assert(rs.next())
-        assert(rs.getString(1) === UTF8String.fromBytes(NumberConverter.toBinary(49960)).toString)
+        assertResult(Array(0, 0, 65533, 40).map(_.toChar)) {
+          rs.getString(1).toCharArray
+        }
```

Comment:
@dongjoon-hyun @HyukjinKwon Maybe we should uses CharArray.

Reply:
What is the difference? Could you tell us the root cause you think?

Reply:
I'm not sure. May be related to environment:
https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/4832/
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109194/testReport/

These two PullRequestBuilders are built on the same machine(amp-jenkins-worker-03). The first one failed, but the last one successful. The environment differences:
![image](https://user-images.githubusercontent.com/5399861/63208095-c7effd00-c102-11e9-8542-53cfad79716c.png)

Reply:
I will verify it later.

Reply:
Maybe, something like `LANG=en_US.UTF_8`?

Reply:
Yes. I can reproduce it by `unset LANG`:
```
[info] - HIVE_CLI_SERVICE_PROTOCOL_V1 get binary type *** FAILED *** (458 milliseconds)
[info]   "[?](" did not equal "[?](" (SparkThriftServerProtocolVersionsSuite.scala:227)
```

Reply:
@shaneknapp I think we need to standardize `LANG` across the workers? or should I try giving it shot?
That seems reasonable though I am distantly concerned that something in the code relies on this setting. That said, I think it's probably correct to set `LANG` in a prod environment and correct to set it to this value anyway.

Reply:
> I'm not sure. May be related to environment:
> https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/4832/
> https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109194/testReport/
> 
> These two PullRequestBuilders are built on the same machine(amp-jenkins-worker-03). The first one failed, but the last one successful. The environment differences:
> ![image](https://user-images.githubusercontent.com/5399861/63208095-c7effd00-c102-11e9-8542-53cfad79716c.png)

ok, the SparkPullRequestBuilder is triggered by the amplab jenkins bot, and the NewSparkPullRequestBuilder is triggered by databricks' test site (http://spark-prs.appspot.com).

i can easily add the LANG variable in the former build.  another option is to add it to the default environment vars for all workers.

Reply:
(fyi, i just added LANG=en_US.UTF-8 to the NewSparkPullRequestBuilder job)

Reply:
@shaneknapp I'd set it everywhere, personally, if it's easy.

Reply:
it's easy, but requires each worker to be disconnected/reconnected...  i'll get that set up now but it won't go in to effect until i have the opportunity to restart things.

Reply:
Thank you @shaneknapp

Reply:
Could we close this PR?

Reply:
Thank you for investigating this so far, @wangyum . Yes. It looks like that. 
You can close this and the JIRA together. If the failure is reported again, we can reopen this.